_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,164 @@
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_3.png" alt="Release Save Version Step" title="Release Save Version Step" class="img img-responsive style-screengrab">
+
+## Deploy 
+
+In the previous step we created an image which we can use now to deploy our application. We are going to create another build configuration: **Deploy**. This build will run a Docker container on the **TeamCity** agent based on the image from the **Release** phase.
+
+Our build configuration must contain three elements. The first one is a trigger. Again, we use the **Finish Build Trigger** with a dependency to the **Release** build configuration.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_trigger.png" alt="Deploy Trigger" title="Deploy Trigger" class="img img-responsive style-screengrab">
+
+The second element is a version string of our image. We can obtain this information from the **artifact** created by the **Release** build. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_artifact.png" alt="Deploy Artifact" title="Deploy Artifact" class="img img-responsive style-screengrab">
+
+The last step is a little bit more complicated. Before we deploy the new version of the application the deployment script need to check if the container with the previous version is already deployed on the machine. If we have such a container we need to stop and remove it. 
+Next we read the current version from the **artifact** and create the new container based this version.  
+
+```
+if [ -n "$(docker ps --filter label=simple_application -q)" ]; then 
+   docker stop simple_application
+   docker rm simple_application
+fi;
+
+version_to_deploy=$(cat image.version)
+docker run -d -p 80:4567 \
+           --name simple_application \
+           --label simple_application \
+           registry.private/simple_application:${version_to_deploy}
+```
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_step.png" alt="Deploy Step" title="Deploy Step" class="img img-responsive style-screengrab">
+
+The Continuous Deployment pipeline is ready. Now every change in the master branch of our repository will build, release and deploy our application.   
+
+## It's done.
+
+The continuous deployment pipeline described in this post is of course simplified. Between our **Release** and **Deploy** steps we would like to do some 
+additional tests on a production-like environment or introduce [zero downtime deployments](http://codurance.com/services/training/devops-training/), but our approach to any deployment should remain unchanged. 
+
+Use the **Docker** image. You can ensure the consistent execution environment for your application on every stage of the Continuous Deployment. Now you decide how the code is executed and you cannot blame admins anymore for installing a wrong version of Java or Ruby. 
+
+You control your application.
```

Comment:
@sandromancuso 
I depends how you look at an application. If you think only about the source code that is true. You control you application. 
But if you don't control how your application is executed you don't have control at all. You can run it locally and everything will be fine, but on production everything is broken, because someone installed oracle JDK instead of OpenJDK. 

Probably I will remove that sentence.